### ISSUE TITLE
Quarto fails during rendering of HTML

### ISSUE THREAD

**Issue by MareikeJaniak** [sigven/cpsr#60]
Hi!

We're excited about the new versions of cpsr and pcgr, thanks so much for developing and maintaining these useful tools!

I am running into an error with quarto when running cpsr with the example data. 

My command: 

`cpsr --force_overwrite --secondary_findings --gwas_findings --panel_id 1 --input_vcf ~/scratch/cpsr/inst/examples/example.vcf.gz --refdata_dir $PCGR_DATA --vep_dir $PCGR_VEP_CACHE --output_dir pairedVariants/ensemble/tumorPair_COLO829/cpsr --genome_assembly grch37 --sample_id example --debug`

The error: 

```
output file: cpsr_report_sample.knit.md

Warning messages:
1: In file.append(to[okay], from[okay]) : write error during file append
2: In file.append(to[okay], from[okay]) : write error during file append
3: In file.append(to[okay], from[okay]) : write error during file append
4: In file.append(to[okay], from[okay]) : write error during file append
ERROR: Theme file compilation failed:

Loading failed: Expected little-endian ELF object.

Stack trace:

Loading failed: Expected little-endian ELF object.
    at dartCommand (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:42801:15)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async dartCompile (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:42763:5)
    at async compileWithCache (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:42992:17)
    at async compileSass (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:42853:12)
    at async resolveSassBundles (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:65203:27)
    at async resolveExtras (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:69229:18)
    at async runPandoc (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:68743:24)
    at async renderPandoc (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:70041:26)
    at async Object.onRender (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:77261:36)
Error in `quarto::quarto_render()`:
✖ Error running quarto cli.
Caused by error:
! System command 'quarto' failed
Backtrace:
    ▆
 1. └─cpsr::write_cpsr_output(cps_report, output_format = "html")
 2.   └─quarto::quarto_render(...)
 3.     └─quarto:::quarto_run(args, echo = TRUE, quarto_bin = quarto_bin)
 4.       └─base::tryCatch(...)
 5.         └─base (local) tryCatchList(expr, classes, parentenv, handlers)
 6.           └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
 7.             └─value[[3L]](cond)
 8.               └─cli::cli_abort(msg, call = .call, parent = e)
 9.                 └─rlang::abort(...)
Execution halted
```

In case it's relevant, we are using pcgr in an HPC environment where it is installed in a read-only shared file system. 

Please let me know if there is any additional information I can provide!

Best, 
Mareike

**Comment by sigven** [comment]
Dear Mareike,

Thanks for reporting on this. I have a suspicion it could be related to how quarto is able to operate in your HPC environment, see this issue:

https://github.com/quarto-dev/quarto-cli/issues/6651

I'll have to dig a bit more to understand this, sorry about the inconvience. Have you tried running with Apptainer/Singularity? 

best,
Sigve

**Comment by sigven** [comment]
Maybe test `quarto check` first and see if that gives you an error?

**Comment by MareikeJaniak** [comment]
Hi Sigve, 
Thanks so much for the quick reply!

I've tried the following:
```[mcj43@bc11234 ~]$ source /cvmfs/soft.mugqic/root/software/pcgr/pcgr-2.0.0/envs/pcgrr/etc/conda/activate.d/quarto.sh
[mcj43@bc11234 ~]$ quarto check
Quarto 1.4.550
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
ERROR: TypeError: Invalid Version: Dart SDK version: 2.18.2 (stable) (Tue Sep 27 13:24:11 2022 +0200) on "linux_x64"

Stack trace:
    at new SemVer (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:27022:19)
    at Range.test (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:27432:23)
    at satisfies (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:27649:18)
    at checkVersion (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:98444:14)
    at checkVersions (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:98470:5)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async check (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:98424:13)
    at async Command.fn (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:98710:5)
    at async Command.execute (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:8104:13)
    at async quarto (file:///cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/quarto.js:114999:5)```

Not sure if that suggests any paths forward?

**Comment by pdiakumis** [comment]
That's so strange. Mareike, is `pcgr-2.0.0/envs/pcgrr/bin/sass --version` also giving you `2.18.2`?

This is what the Docker image is giving (I have v1.4.1.9015 locally but should be the same as v2.0.0): 

<details>
<summary>Click</summary>

```
quarto check
Quarto 1.4.550
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.58.3: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.550
      Path: /opt/mambaforge/envs/pcgrr/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Tex:  (not detected)

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.3 (Conda)
      Path: /opt/mambaforge/envs/pcgrr/bin/python
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with conda install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /opt/mambaforge/envs/pcgrr/lib/R
      LibPaths:
        - /opt/mambaforge/envs/pcgrr/lib/R/library
      knitr: 1.47
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK
```
</details>

The Dart Sass version is the same as the conda lockfile at https://github.com/sigven/pcgr/blob/main/conda/env/lock/pcgrr-linux-64.lock#L8

**Comment by MareikeJaniak** [comment]
Yes, I'm getting the same version:

```
[mcj43@bc11234 ~]$ /cvmfs/soft.mugqic/CentOS6/software/pcgr/pcgr-2.0.0/envs/pcgrr/bin/sass --version
Dart SDK version: 2.18.2 (stable) (Tue Sep 27 13:24:11 2022 +0200) on "linux_x64"
```

**Comment by pdiakumis** [comment]
I'm flying completely blind here but maybe you could try replacing that `sass` binary with the one from <https://github.com/sass/dart-sass/releases/tag/1.58.3>. Under assets I think you'll need the `dart-sass-1.58.3-linux-x64.tar.gz` one. Decompress and there should be a `sass` binary in the `dart-sass` folder. Let's see if that does anything.

**Comment by MareikeJaniak** [comment]
Hi, 
I just wanted to let you know that we've resolved this issue by installing version 2.0.1. I'm not totally sure if the wrong `sass` binary was in the 2.0.0 release or if it somehow ended up in our pcgr install because of an existing installation, but either way 2.0.1 is now working for us with the example data. 

Thanks for your help!
Best, 
Mareike
